libvmi/driver/xen/xen_events.c

```diff
@@ -862,7 +862,10 @@ status_t xen_set_mem_access(vmi_instance_t vmi, mem_access_event_t *event,
     dbprint(VMI_DEBUG_XEN, "--Setting memaccess for domain %"PRIu64" on physical address: %"PRIu64" npages: %"PRIu64"\n",
         dom, event->physical_address, npages);
 
-    rc = xc_set_mem_access(xch, dom, access, page_key, npages);
+    if ( !altp2m_idx )
```

Comment:
I'm pretty sure 0 is a valid altp2m index.  I think this needs some other value (-1?) or an additional flag based on the presence of `VM_EVENT_FLAG_ALTERNATE_P2M`.

Unless xc_set_mem_access() is equivalent to `xc_altp2m_set_mem_access(...,..., 0, ..., ...)`, which is unclear.  I'm not finding these APIs very well documented.